src/config/mod.rs

```diff
@@ -34,8 +34,14 @@ pub(crate) struct Settings {
 impl Settings {
     /// Assemble runtime settings.
     pub(crate) fn assemble() -> Fallible<Self> {
-        let dirs = vec!["/usr/lib", "/run", "/etc"];
-        let cfg = inputs::ConfigInput::read_configs(&dirs, crate_name!())?;
+        let prefixes = vec![
+            "/usr/lib/".to_string(),
+            "/run/".to_string(),
+            "/etc/".to_string(),
+        ];
+        let common_path = format!("{}/config.d/", crate_name!());
+        let extensions = vec!["toml".to_string()];
+        let cfg = inputs::ConfigInput::read_configs(prefixes, &common_path, extensions)?;
```

Comment:
Ahh - having to later borrow the string returned by `format!` makes me wonder if `liboverdrop::FragmentScanner::new()` should accept an owned `String` for the `shared_path` parameter instead of `&str`. This would be consistent with the `prefixes` and `extensions` holding `String` as well.

Having `shared_path: &str` is easier for directly passing in a slice e.g. `"my-app/config.d"` but the more common pattern would probably be using `format!` and passing in the crate name.

Reply:
Opened an issue for this: https://github.com/overdrop/liboverdrop-rs/issues/7